firefox-ios/Client/TabManagement/Legacy/LegacyTabManager.swift

```diff
@@ -232,7 +232,8 @@ class LegacyTabManager: NSObject, FeatureFlaggable, TabManager, TabEventHandler
     }
 
     // MARK: Get tabs
-    func getTabFor(_ url: URL) -> Tab? {
+    func getTabFor(_ url: URL, reversed: Bool = false) -> Tab? {
+        let tabs = reversed ? self.tabs.reversed() : self.tabs
```

Comment:
If we have multiple tabs with the same URL, it will return the first one it finds. According to @nbhasin2 we need to select the latest tab with the same URL so that's why the reverse is needed.